src/common.rs

```diff
@@ -28,6 +29,31 @@ pub struct MessageHeader {
     pub flags: u8,
 }
 
+impl MessageHeader {
+    pub(crate) fn from_buf(buf: impl AsRef<[u8]>) -> Self {
```

Comment:
Would `From`/`Into` traits implementation be more applicable here?

```rust
impl<T> From<T> for MessageHeader where T: AsRef<[u8]> {}` 
```

Reply:
Sure, fixed.